### ISSUE TITLE
use PyWebScrapBook & WebScrapBook to set focus on an existing tab with a captured URL or open a new tab

### ISSUE THREAD

**Issue by uhu53** [danny0838/webscrapbook#206]
I am trying to get smart about a topic (Color Perception, Color Models and -Spaces, Mapping Functions etc.) by structuring it in a MindMap tool (freeplane). I would like to link wikipedia articles & other to certain nodes. When I click on such a link, the corresponding article -- previously captured & decorated with WSB -- should open in Firefox. If the requested article & tab is already open in Firefox, it should set the focus in Firefox to that tab. If it's not already open, it should open the requested page in Firefox in a new tab.

In the MindMap tool (freeplane) I can attach 'URL hyperlinks' to nodes. When I set such a hyperlink to an address like: 'http://localhost:8080/20210225155806951/index.html' (the pyWebScrapBook server) 
it opens in Firefox the requested page (captured and decorated with WSB) **ALWAYS IN A NEW TAB**, albeit I might have opened the concerning page & tab previously by clicking the concerning node in WSB's sidepane . 
(In that new tab I have the full edit functionality of WSB, what I like.)

How can I get the PyWebScrapBook server and WSB to set focus to an existing tab containing the requested content by addressing PyWebScrapBook server?
If the concerning tab does not exist already, it shall open in a new tab -- as is? Is this feasable and how?

I am currently able to debug WSB in Firefox up to a certain degree, but I am new to Browser Extensions (and Javascript) -- I am not new to programming and structured content (professionaly 20yrs+, Perl, Oracle, Content Mgnt, SGML, ...). Can you give me a hint please?

**Comment by danny0838** [comment]
Focusing on the existing tab when clicking a link is likely beyond WSB's core functionality and may conflict with a tab management add-on like Simple Tab Groups. I think you should consult them, or maybe write a standalone add-on for this purpose instead.

**Comment by uhu53** [comment]
thanks danny.
I'm about to dive into the code, server and WSB, struggling to get a picture of its structure. For the server I try pyreverse for the server & delve into WSB beginning with Firefox' "Anatomy of an extension". 
'Simple Tab Groups' I do install, might be sort of a revelation.
Can I use Firefox.debug's Log function as a tracing tool to cut some lanes through WSB's functionality? Or is there a better tool?

**Comment by danny0838** [comment]
AFAIK there's not much tool for debugging a browser extension besides the development tool of the browser itself. You may try Chromium besides Firefox, though.

**Comment by uhu53** [comment]
Hi Danny,

I look for a possibility to transpose my annotations in handbooks from
one version of the same handbook to the next.
I have learned that a WSB-re-capture ports annotations up to a certain
extent to the newly recaptured page. Beside too rough changes from one
version to another there is the limitation that WSB-HTML-edits are not
tagged (in the HTML-source). Looking at the source HTML of a page with
WSB-annotations  I have seen that you mark e.g. highlights neatly but
not HTML-edits.
Is there a possibility to mark them too?
A HTML compare tool would then be able to detect also HTML text
changes.

In my former work we had an SGML Compare Tool we used to report changes
between two versions of SGML manuals (aircraft maintenance) up to the
last byte. These manuals had a relativly strict controlled manual
revision process by a DTD and the issuer's manual writing
prescriptions. Of course this is fundamentally different from the much
weaker formal control of HTML manuals. But anyway I expect some sort of
management and control for the (HTML) manuals of open source projects,
in which I'm interested in.

I tried to debug WSB to find out what's the issue with WSB-HTML-edit,
i.e. I have the ugly suspicion that there is a logical reason for not
marking HTML-edit. But I came to no clue so far.

I appreciate very much some hint from your side.

many thanks

Ueli
